src/Services/src/CloudFoundry/CloudFoundryService.cs

```diff
@@ -876,6 +871,16 @@ public DetailedResult CreateManifestFile(string location, AppManifest manifest)
             }
         }
 
+        public string SerializeManifest(AppManifest manifest)
```

Comment:
The role of serialization doesn't seem to belong to CloudFoundryService.  E.g., there may be additional needs for YAML serialization in other features.  So it seems that this function is better suited to dedicated class/classes, perhaps as part of common utils.

The parameter type doesn't need to be specific to AppManifest.  The serializer can accept plain objects, so you just need a signature like `Serialize(object o)`.  Should you need to deserialize, you can use a method signature like `T Deserialize<T>(string s)`.

Finally, you only need one instance of a serializer (or deserializer).  The serialization code doesn't need to maintain any kind of instance state so you can use a static class to implement.  As part of the static constructor you can create your single serializer instance that can be used whenever serialization is needed.

Reply:
Sure, makes sense & abstracting the (de)serialization stuff into its own class will make testing simpler if we can mock those methods 👍

Reply:
wrt `AppManifest` specificity, I like having that argument be restrictive because we're applying custom rules to how app manifests in particular should be serialized:
```
serializer = new SerializerBuilder()
  .WithNamingConvention(CfAppManifestNamingConvention.Instance)
  ConfigureDefaultValuesHandling(DefaultValuesHandling.OmitDefaults)
  Build();
```

In a static serialization class, how would you allow for different serialization rules? Expose 1 serializer for each rule set? 
E.g. 
```
public static class SerializationService
    {
        static ISerializer cfAppManifestSerializer;
        static ISerializer anotherSerializer;

        static SerializationService()
        {
            cfAppManifestSerializer = new SerializerBuilder()
                                .WithNamingConvention(CfAppManifestNamingConvention.Instance)
                                .ConfigureDefaultValuesHandling(DefaultValuesHandling.OmitDefaults)
                                .Build();

            anotherSerializer = new SerializerBuilder()
                                .WithNamingConvention(CamelCaseNamingConvention.Instance)
                                .ConfigureDefaultValuesHandling(DefaultValuesHandling.OmitNull)
                                .Build();
        }
    }
```

Reply:
Good questions.  You could defer impl improvements until you need more than a single rule set.

One impl might be that the ruleset (more specifically, something representing the ruleset) is an additional option to method.  E.g., assume `RuleSet` is an enum ...
```
serializer.Serialize(o, RuleSet.ManifestRuleSet);
```

You serializers are maintained in a dictionary, keyed by RuleSet value.  You look up the serializer when called, and if does not exist you add it to the dictionary.  

Using a dictionary like that is a common pattern, especially in caching.

Reply:
A simpler approach to above takes advantage that you know all rulesets at compile time.  You can initialize your serializer dictionary at construction time.